doc/lua_api.md

```diff
@@ -9664,29 +9663,96 @@ Currently the only workaround is to use `run_at_every_load = true`.
 Tile definition
 ---------------
 
-* `"image.png"`
-* `{name="image.png", animation={Tile Animation definition}}`
-* `{name="image.png", backface_culling=bool, align_style="node"/"world"/"user", scale=int}`
-    * backface culling enabled by default for most nodes
-    * align style determines whether the texture will be rotated with the node
-      or kept aligned with its surroundings. "user" means that client
-      setting will be used, similar to `glasslike_framed_optional`.
-      Note: supported by solid nodes and nodeboxes only.
-    * scale is used to make texture span several (exactly `scale`) nodes,
-      instead of just one, in each direction. Works for world-aligned
-      textures only.
-      Note that as the effect is applied on per-mapblock basis, `16` should
-      be equally divisible by `scale` or you may get wrong results.
-* `{name="image.png", color=ColorSpec}`
-    * the texture's color will be multiplied with this color.
-    * the tile's color overrides the owning node's color in all cases.
-* deprecated, yet still supported field names:
-    * `image` (name)
+Short form:
+
+```lua
+"mymod_image.png"
+-- Equivalent to `{name = "mymod_image.png"}`.
+```
+
+Long form:
+
+```lua
+{
+    name = "mymod_image.png",
+    -- Texture for the tile, see [Textures].
+
+    image = "mymod_image.png",
+    -- Deprecated. Alternative to `name`.
+
+    backface_culling = bool,
+    -- If `true`, faces will only be drawn from one side.
+    --
+    -- Default:
+    -- * `false` for drawtypes `plantlike`, `firelike`, `mesh`, and `liquid`.
+    -- * `false` for `special_tiles` of drawtype `plantlike_rooted`.
+    -- * `true` otherwise.
+
+    tileable_horizontal = bool,
+    tileable_vertical = bool,
+    -- Decides how textures are sampled out-of-bounds.
+    --
+    -- If `true`, pixels are repeated, if `false`, values are clamped to edge.
+    --
+    -- `tileable_horizontal` and `tileable_vertical` apply to texture coordinates
+    -- horizontal and vertical respecitively in uv space, not world space.
+    --
+    -- Default (for both):
+    -- * `false` for drawtypes `plantlike`, and `firelike`.
+    -- * `false` for `special_tiles` of drawtype `plantlike_rooted`.
+    -- * `true` otherwise.
+
+    align_style = "node",
+    -- Determines whether the texture will be rotated with the node or kept
+    -- aligned with its surroundings.
+    --
+    -- The tile for a face of a node is still chosen by the node's rotation. This
+    -- option only applies to the rotation within the face's plane.
+    --
+    -- One of:
+    -- * "node" (default): Rotated with node.
+    -- * "world": Always default rotation.
+    -- * "user": Client setting `world_aligned_mode` will be used, similar to
+    --   `glasslike_framed_optional`.
+    --
+    -- Only supported by drawtypes made of cuboids.
+    --
+    -- Setting a different value in the corresponding `overlay_tiles` tile is not
+    -- supported.
+
+    scale = int,
```

Comment:
assuming we want to stay with lua types
```suggestion
    scale = number,
```

Reply:
I think int(eger) is fine and a bit clearer.

See also: https://docs.luanti.org/for-creators/lua-conventions/#integers for anyone who may be confused.

The 5.1 reference manual also uses the term integer, e.g. https://www.lua.org/manual/5.1/manual.html#pdf-math.random.

Reply:
We also use "int" elsewhere in the doc.